### PR TITLE
fix: `missing file parameter` exception [ROADRUNNER-111]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
will fix https://snyksec.atlassian.net/browse/ROADRUNNER-111

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt [hunk 1]
```diff
@@ -30,12 +30,17 @@ class PDU private constructor() : PlatformDependentUtilsBase() {
...
         val psiFile = toPsiFile(file)
         // looks like we don't need ReadAction for this (?)
-        var absolutePath = psiFile.virtualFile.path
+        val absolutePath = psiFile.virtualFile.path
         val projectPath = psiFile.project.basePath
-        if (projectPath != null) {
-            absolutePath = absolutePath.replace(projectPath, "")
+            ?: throw IllegalStateException("No Project base Path found for file $psiFile")
+        // `ignoreCase = true` needed due to https://youtrack.jetbrains.com/issue/IDEA-268081
+        val projectBasedPath = absolutePath.replaceFirst(projectPath, "", ignoreCase = true)
```
Comment (aldanchenko) [inline src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt:37]: Just curious. Could this produce unexpected issues if user OS is case sensitive and he has, for example,

/home/user/projects/testproj/
/home/user/projects/TestProj/

?

**Review by aldanchenko — APPROVED** [review]
LGTM. Great job Artsiom! 👍

File: src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt (hunk 1 above)
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt:37]: It should not, as we comparing projectPath against projectPathWithFilePathInside, so technically speaking every time when those comparison fail is an Exception (that's why we have IllegalStateException and not just log.warn)

Comment (aldanchenko) [reply src/main/kotlin/io/snyk/plugin/snykcode/core/PDU.kt:37]: Great work! Thanks for clarifications! 👍

**Pull request merged**